pandaserver/daemons/scripts/task_evaluator.py

```diff
@@ -46,41 +46,34 @@ def _wrapped_method(self, *args, **kwargs):
                     _method(self, *args, **kwargs)
                 except Exception as exc:
                     pass
+
             return _wrapped_method
+
         return _decorator(method)
 
     def update(self, metric, entity_dict):
-        tmp_log = logger_utils.make_logger(main_logger, 'TaskEvaluationDB.update')
-        tmp_log.debug('start metric={}'.format(metric))
+        tmp_log = logger_utils.make_logger(main_logger, "TaskEvaluationDB.update")
+        tmp_log.debug("start metric={}".format(metric))
         # sql
-        sql_query_taskid = (
-            """SELECT jediTaskID """
-            """FROM ATLAS_PANDA.Task_Evaluation """
-            """WHERE metric = :metric """
-        )
+        sql_query_taskid = """SELECT jediTaskID """ """FROM ATLAS_PANDA.Task_Evaluation """ """WHERE metric = :metric """
         sql_update = (
             """UPDATE ATLAS_PANDA.Task_Evaluation SET """
-                """value_json = :patch_value_json, """
-                """timestamp = :timestamp """
+            """value_json = :patch_value_json, """
+            """timestamp = :timestamp """
             """WHERE jediTaskID=:taskID AND metric=:metric """
         )
-        sql_insert = (
-            """INSERT INTO ATLAS_PANDA.Task_Evaluation """
-                """VALUES ( """
-                    """:taskID, :metric, :patch_value_json, :timestamp """
-                """) """
-        )
+        sql_insert = """INSERT INTO ATLAS_PANDA.Task_Evaluation """ """VALUES ( """ """:taskID, :metric, :patch_value_json, :timestamp """ """) """
```

Comment:
I don't like the way Black "fixes" these intentional indentations and newlines of long strings for readability (typically for sql). Maybe my code style here is not the best, but every collapsing everything down to a single line looks bad.

BTW, I tend to assign `sql = (a multiline long string, )` in one go, instead of repeated string appending `sql += ...` `sql += ...` . Sometimes I have to use triple quotes for sql when some placeholder can get JSON (which contains quotes).

Reply:
My preference is a multiline triple quoted string, like:
```
            sql = """
                   SELECT /* use_json_type */ panda_queue, data, b.site_name, c.role
                   FROM (ATLAS_PANDA.schedconfig_json a
                   LEFT JOIN ATLAS_PANDA.panda_site b ON a.panda_queue = b.panda_site_name)
                   LEFT JOIN ATLAS_PANDA.site c ON b.site_name = c.site_name
                   WHERE panda_queue IS NOT NULL
                   """
```
TBH I don't like `sql  = (a multiline long string)` (or what black did with it) and my least favorite option is `sql+=...` (I find it very difficult to read for long queries).

Reply:
Probably, this is a matter of preference, but queries with multiline triple-quoted strings are shown as multiple lines in the log files, which makes log-parsing with grep etc a bit complicated. Sometimes, we want to see raw SQL statements and bind-variables if they are constructed dynamically.

Reply:
Would you be satisfied with removing newlines when logging sql?
```
>>> sql = """
... line1
... line2
... line3
... """
>>> print(sql.replace('\n', ' '))
 line1 line2 line3
>>> print(sql)

line1
line2
line3
```

Reply:
How about creating a class for sql command, so we can instantiate it with Fernando's style, and we define its `__str__` or an attribute to be the replaced compact string which can be feed in execute and logger?
The class can be extend for extra features in the future (e.g. making the list and var_map of placeholders filled in IN clause (a0, a1, ...). Current we have to program that everywhere we need it)

Reply:
@mightqxc Is [this](https://github.com/PanDAWMS/panda-server/pull/249/commits/1d8d4e9999c6082c727cf4cdef67bfecceced88b) what you meant?

Reply:
Yeah it looks nice.